File: src/types/checkoutRequest.ts
export interface CheckoutRequest {
    actualCheckOutTime: string;
    notes?: string;
    additionalCharges?: number;
    paymentMethod?: string;
    isEarlyCheckout: boolean;
    feedback?: string;
}

export interface CheckoutRequestCreate {
    notes?: string;
    additionalCharges?: number;
    paymentMethod?: string;
    isEarlyCheckout?: boolean;
    feedback?: string;
}

export type CheckoutRequestUpdate = Partial<CheckoutRequestCreate>; 

File: SmartStayHaven.API/Models/CheckoutRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartStayHaven.API.Models
{
    public class CheckoutRequest
    {
        public DateTime ActualCheckOutTime { get; set; } = DateTime.UtcNow;

        public string? Notes { get; set; }

        public decimal? AdditionalCharges { get; set; }

        public string? PaymentMethod { get; set; }

        public bool IsEarlyCheckout { get; set; }

        public string? Feedback { get; set; }
    }
} 